src/lib/Microsoft.Health.Fhir.Ingest/Telemetry/Metrics/IomtMetrics.cs

```diff
@@ -0,0 +1,278 @@
+                { _categoryDimension, Category.Latency },
+                { _stageDimension, ConnectorStage.Normalization },
+            });
+
+        private static Metric _patientDeviceMismatch = new Metric(
+            "PatientDeviceMismatchException",
+            new Dictionary<string, object>
+            {
+                { _nameDimension, "PatientDeviceMismatchException" },
+                { _categoryDimension, Category.Errors },
+                { _errorTypeDimension, ErrorType.FHIRResourceError },
+                { _errorSeverityDimension, ErrorSeverity.Warning },
+                { _stageDimension, ConnectorStage.FHIRConversion },
+            });
+
+        private static Metric _notSupportedException = new Metric(
+            "NotSupportedException",
+            new Dictionary<string, object>
+            {
+                { _nameDimension, "NotSupportedException" },
+                { _categoryDimension, Category.Errors },
+                { _errorTypeDimension, ErrorType.FHIRResourceError },
+                { _errorSeverityDimension, ErrorSeverity.Warning },
+                { _stageDimension, ConnectorStage.FHIRConversion },
+            });
+
+        private static Metric _multipleResourceFoundException = new Metric(
+           "MultipleResourceFoundException",
+           new Dictionary<string, object>
+           {
+                { _nameDimension, "MultipleResourceFoundException" },
+                { _categoryDimension, Category.Errors },
+                { _errorTypeDimension, ErrorType.FHIRResourceError },
+                { _errorSeverityDimension, ErrorSeverity.Warning },
+                { _stageDimension, ConnectorStage.FHIRConversion },
+           });
+
+        private static Metric _templateNotFoundException = new Metric(
+           "TemplateNotFoundException",
+           new Dictionary<string, object>
+           {
+                { _nameDimension, "TemplateNotFoundException" },
+                { _categoryDimension, Category.Errors },
+                { _errorTypeDimension, ErrorType.GeneralError },
+                { _errorSeverityDimension, ErrorSeverity.Critical },
+                { _stageDimension, ConnectorStage.Unknown },
+           });
+
+        private static Metric _correlationIdNotDefinedException = new Metric(
+           "CorrelationIdNotDefinedException",
+           new Dictionary<string, object>
+           {
+                { _nameDimension, "CorrelationIdNotDefinedException" },
+                { _categoryDimension, Category.Errors },
+                { _errorTypeDimension, ErrorType.DeviceMessageError },
+                { _errorSeverityDimension, ErrorSeverity.Critical },
+                { _stageDimension, ConnectorStage.Normalization },
+           });
+
+        /// <summary>
+        /// The latency between event ingestion and output to FHIR processor.
+        /// </summary>
+        public static Metric MeasurementIngestionLatency()
```

Comment:
I was thinking we may wanted to add to the Metric later for PaaS if we wanted to add more dimensions in PaaS that aren't applicable in OSS.

Reply:
But the method still returns a Metric object.  It's static so it can't be overridden.  Not following how the extension would be different if this is property instead of a method.  Could you clarify please?

Reply:
I was hoping to add some dimensions to the Metric object before we log the metric. 

So for example if we had a DeviceEventProcessingLatency metric, we currently have 3 dimensions (name, category, stage), but let's say we want to add a 4th dimension for internal use which isn't applicable to OSS. Right before logging the metric I was hoping to modify the metric object dimensions by setting a 4th dimension key/value if it did not already exist.

I'm not positive on this, but thought that making the metric readonly would prevent us from being able to modify the metric object.